python/ray/dag/compiled_dag_node.py

```diff
@@ -150,54 +138,58 @@ def _wrap_exception(exc):
     return wrapped
 
 
-def _exec_task(self, task: "ExecutableTask", idx: int) -> bool:
+def _select_next_nodes(
+    actor_to_candidates: Dict["ray.actor.ActorHandle", List[DAGOperationGraphNode]],
```

Comment:
QQ: should we use actor_id instead of actor handle for key? kind of weird we are using actor handle for the key..

Reply:
I just follow the convention of the existing codebase. For example,

* https://github.com/ray-project/ray/blob/b299b0bf2be6f9c685aec970b15677b588254aff/python/ray/dag/compiled_dag_node.py#L519
* https://github.com/ray-project/ray/blob/b299b0bf2be6f9c685aec970b15677b588254aff/python/ray/dag/compiled_dag_node.py#L522
* https://github.com/ray-project/ray/blob/b299b0bf2be6f9c685aec970b15677b588254aff/python/ray/dag/compiled_dag_node.py#L526

I will use `actor_id` for this function in this PR, and update the rest in a follow up PR https://github.com/ray-project/ray/issues/47012.

Reply:
Updated https://github.com/ray-project/ray/pull/46911/commits/50ff5c1385be55a1220fd8b9ef05f28b89444c03